lib/pure/os.nim

```diff
@@ -149,8 +150,9 @@ proc findExe*(exe: string, followSymlinks: bool = true;
     checkCurrentDir()
   let path = string(getEnv("PATH"))
   for candidate in split(path, PathSep):
+    if candidate.len == 0: continue
     when defined(windows):
-      var x = (if candidate[0] == '"' and candidate[^1] == '"':
+      var x = (if candidate.len > 2 and candidate[0] == '"' and candidate[^1] == '"':
```

Comment:
What about ""?

Reply:
@koranza 
No.
Check this:
```nim
var candidate = ""
echo candidate[0] == '"' and candidate[^1] == '"'
```
and
```nim
var candidate = ""
echo candidate.len > 2 and candidate[0] == '"' and candidate[^1] == '"'
```

Reply:
I thought this was necessary to avoid IOB errors for candidate[^1], substr, and '\' operations, but these will only throw errors if candidate's length is 0, which we already check for